src/HttpClient.php

```diff
@@ -224,14 +272,29 @@ protected function post($uri, array $options = [])
     }
 
     /**
-     * Post the given data to Bugsnag in json form.
+     * Deliver the given events to the notification API.
      *
      * @param string $uri  the uri to hit
      * @param array  $data the data send
      *
      * @return void
+     *
+     * @deprecated Use {HttpClient::deliverEvents} instead
      */
     protected function postJson($uri, array $data)
+    {
+        $this->deliverEvents($uri, $data);
+    }
+
+    /**
+     * Deliver the given events to the notification API.
+     *
+     * @param string $uri  the uri to hit
+     * @param array  $data the data send
+     *
+     * @return void
+     */
+    protected function deliverEvents($uri, array $data)
```

Comment:
The naming of the event delivery functions is a little different to sessions and builds. Once we remove the deprecations, we'll have `sendBuildReport(array $data)`, `sendSession(array $data)` but `sendEvents()` and `deliverEvents($uri, array $data)`.

As we're making deprecations, I think we should aim towards `sendEvents(array $data)`. This may mean you need a lower-level private function that takes an array and URI which you can use for `postJson`. Then switch naming around so that `deliverEvents` is the parameterless method that reads the queue and is called from `send` until that is deprecated.

Reply:
Are you saying to move the queued events somewhere else? I don't think it makes sense to make `sendEvents` take the array of events, unless `HttpClient` doesn't track the queue of events itself — otherwise the thing calling `sendEvents` has to read the queue from `HttpClient` first, then pass that back to `HttpClient` to send

Reply:
It's just a naming thing - we have `sendXXX` elsewhere that takes a param with what to send. Whereas for events it's called `deliverEvents` that takes the parameter and `sendEvents` reads the queue. Was just suggesting to reverse the naming.

Reply:
Wouldn't that make the public interface of `HttpClient` `sendSessions`, `sendBuildReport` and `deliverEvents` though?

I think it'd be nice to move the queue elsewhere and make the `sendX` methods all take some data to send, but maybe in a separate PR?